src/OSSupport/GetAddressInfoError.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+
+
+/** Returns the readable form of a getaddressinfo type error code */
+inline AString ErrorString(int a_ErrorCode)
+{
+	// Note gai_strerror is not threadsafe on windows
+	#ifdef _WIN32
+		AString ErrorStr(GAI_STRERROR_BUFFER_SIZE, '\0');
+
+		int MsgLen = FormatMessageA(
+			FORMAT_MESSAGE_FROM_SYSTEM |
+			FORMAT_MESSAGE_IGNORE_INSERTS |
+			FORMAT_MESSAGE_MAX_WIDTH_MASK,
+			nullptr,
+			a_ErrorCode,
+			MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+			&ErrorStr[0],
```

Comment:
As far as I can tell, that doesn't guarantee that the pointer will stay valid after the statement (the assignment, or in the case of cppreference, the function call) is executed.
In my opinion it would be legal for a `std::string` implementation to return a temporary object that has a copy of the contained string including the NUL terminator, has `const char *` conversion defined on it, so that it is possible to read it, and in its destructor frees the copy of the string - it would still pass as a valid `std::string` but will not work with your (our) code. None of the implementations I know of currently do that, but it is possible for some obscure STL implementation to try it (for example to implement true copy-on-write strings, what do I know).